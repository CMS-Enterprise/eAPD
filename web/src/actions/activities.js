import { saveApd } from './apd';
import { updateBudget } from './budget';

export const ADD_ACTIVITY = 'ADD_ACTIVITY';
export const ADD_ACTIVITY_CONTRACTOR = 'ADD_ACTIVITY_CONTRACTOR';
export const ADD_ACTIVITY_DIRTY = 'ADD_ACTIVITY_DIRTY';
export const ADD_ACTIVITY_GOAL = 'ADD_ACTIVITY_GOAL';
export const ADD_ACTIVITY_EXPENSE = 'ADD_ACTIVITY_EXPENSE';
export const ADD_ACTIVITY_MILESTONE = 'ADD_ACTIVITY_MILESTONE';
export const ADD_ACTIVITY_STATE_PERSON = 'ADD_ACTIVITY_STATE_PERSON';
export const EXPAND_ACTIVITY_SECTION = 'EXPAND_ACTIVITY_SECTION';
export const REMOVE_ACTIVITY = 'REMOVE_ACTIVITY';
export const REMOVE_ACTIVITY_CONTRACTOR = 'REMOVE_ACTIVITY_CONTRACTOR';
export const REMOVE_ACTIVITY_GOAL = 'REMOVE_ACTIVITY_GOAL';
export const REMOVE_ACTIVITY_EXPENSE = 'REMOVE_ACTIVITY_EXPENSE';
export const REMOVE_ACTIVITY_MILESTONE = 'REMOVE_ACTIVITY_MILESTONE';
export const REMOVE_ACTIVITY_STATE_PERSON = 'REMOVE_ACTIVITY_STATE_PERSON';
export const TOGGLE_ACTIVITY_CONTRACTOR_HOURLY =
  'TOGGLE_ACTIVITY_CONTRACTOR_HOURLY';
export const TOGGLE_ACTIVITY_SECTION = 'TOGGLE_ACTIVITY_SECTION';
export const UPDATE_ACTIVITY = 'UPDATE_ACTIVITY';

const actionWithYears = (type, other) => (dispatch, getState) =>
  dispatch({ type, ...other, years: getState().apd.data.years });

export const addActivity = () => (dispatch, getState) => {
  const oldActivities = new Set(getState().activities.allKeys);

  dispatch(actionWithYears(ADD_ACTIVITY));
  dispatch(updateBudget());

  const newKey = getState().activities.allKeys.find(k => !oldActivities.has(k));

  dispatch({
    type: ADD_ACTIVITY_DIRTY,
    data: getState().activities.byKey[newKey]
  });
};

export const updateActivity = (key, updates, isExpense = false) => dispatch => {
  dispatch({
    type: UPDATE_ACTIVITY,
    key,
    updates
  });
  if (isExpense) {
    dispatch(updateBudget());
  }
};

export const addActivityContractor = (key, { save = saveApd } = {}) => async (
  dispatch,
  getState
) => {
  const activity = getState().activities.byKey[key];
  const oldIds = activity.contractorResources.map(c => c.id);

  dispatch(actionWithYears(ADD_ACTIVITY_CONTRACTOR, { key }));

  const newApd = await dispatch(save());
  const newContractor = newApd.activities
    .find(a => a.id === activity.id)
    .contractorResources.find(c => !oldIds.includes(c.id));

  const updates = {
    contractorResources: {
      [activity.contractorResources.length]: { id: newContractor.id }
    }
  };

  dispatch(updateActivity(key, updates));
};

export const addActivityGoal = key => ({ type: ADD_ACTIVITY_GOAL, key });

export const addActivityExpense = key =>
  actionWithYears(ADD_ACTIVITY_EXPENSE, { key });

export const addActivityMilestone = key => ({
  type: ADD_ACTIVITY_MILESTONE,
  key
});

export const addActivityStatePerson = key =>
  actionWithYears(ADD_ACTIVITY_STATE_PERSON, { key });

export const expandActivitySection = key => ({
  type: EXPAND_ACTIVITY_SECTION,
  key
});

export const removeActivity = (key, { global = window } = {}) => dispatch => {
  if (global.confirm('Do you really want to delete this activity?')) {
    dispatch({ type: REMOVE_ACTIVITY, key });
    dispatch(updateBudget());
  }
};

export const removeActivityContractor = (
  key,
  contractorKey,
  { global = window } = {}
) => dispatch => {
  if (
    global.confirm('Do you really want to delete this contractor resource?')
  ) {
    dispatch({
      type: REMOVE_ACTIVITY_CONTRACTOR,
      key,
      contractorKey
    });
    dispatch(updateBudget());
  }
};

export const removeActivityGoal = (
  key,
  goalKey,
  { global = window } = {}
) => dispatch => {
  if (global.confirm('Do you really want to delete this activity goal?')) {
    dispatch({
      type: REMOVE_ACTIVITY_GOAL,
      key,
      goalKey
    });
  }
};

export const removeActivityExpense = (
  key,
  expenseKey,
  { global = window } = {}
) => dispatch => {
  if (global.confirm('Do you really want to delete this non-personnel cost?')) {
    dispatch({
      type: REMOVE_ACTIVITY_EXPENSE,
      key,
      expenseKey
    });
    dispatch(updateBudget());
  }
};

export const removeActivityMilestone = (
  key,
  milestoneKey,
  { global = window } = {}
) => dispatch => {
  if (global.confirm('Do you really want to delete this activity milestone?')) {
    dispatch({
      type: REMOVE_ACTIVITY_MILESTONE,
      key,
      milestoneKey
    });
  }
};

export const removeActivityStatePerson = (
  key,
  personKey,
  { global = window } = {}
) => dispatch => {
  if (global.confirm('Do you really want to delete this personnel entry?')) {
    dispatch({
      type: REMOVE_ACTIVITY_STATE_PERSON,
      key,
      personKey
    });
    dispatch(updateBudget());
  }
};

export const toggleActivitySection = key => ({
  type: TOGGLE_ACTIVITY_SECTION,
  key
});

export const toggleActivityContractorHourly = (
  key,
  contractorKey,
  useHourly
) => dispatch => {
  dispatch({
    type: TOGGLE_ACTIVITY_CONTRACTOR_HOURLY,
    key,
    contractorKey,
    useHourly
  });
  dispatch(updateBudget());
};
