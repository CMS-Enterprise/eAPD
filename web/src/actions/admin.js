import { notify } from './notification';
import axios from '../util/api';

export const ADMIN_CREATE_USER_REQUEST = Symbol(
  'admin : create user : request'
);
export const ADMIN_CREATE_USER_SUCCESS = Symbol(
  'admin : create user : success'
);
export const ADMIN_CREATE_USER_ERROR = Symbol('admin : create user : error');

export const createUser = ({ email, name, password, state }) => dispatch => {
  dispatch({ type: ADMIN_CREATE_USER_REQUEST });

  return axios
    .post('/users', { email, name, password, state })
    .then(() => {
      dispatch({ type: ADMIN_CREATE_USER_SUCCESS });
      dispatch(notify('User created!'));
    })
    .catch(e => {
      dispatch({ type: ADMIN_CREATE_USER_ERROR });

      switch (e.response.data.error) {
        case 'add-user-invalid':
          dispatch(notify('Error: Email and password are required'));
          break;
        case 'add-user-email-exists':
          dispatch(notify('Error: A user with this email already exists'));
          break;
        case 'add-user-weak-password':
          dispatch(notify('Error: The provided password is too weak'));
          break;
        case 'add-user-invalid-phone':
          dispatch(notify('Error: The provided phone number is invalid'));
          break;
        default:
          dispatch(notify('Unknown error creating user'));
          break;
      }

      return Promise.reject();
    });
};
