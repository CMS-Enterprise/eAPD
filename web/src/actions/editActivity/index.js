export * from './contractorResources';
export * from './costAllocate';
export * from './goals';
export * from './overview';
