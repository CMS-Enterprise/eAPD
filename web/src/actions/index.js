export const INCREMENT_COUNT = 'INCREMENT_COUNT';
export const DECREMENT_COUNT = 'DECREMENT_COUNT';

export const increment = () => ({ type: INCREMENT_COUNT });
export const decrement = () => ({ type: DECREMENT_COUNT });
