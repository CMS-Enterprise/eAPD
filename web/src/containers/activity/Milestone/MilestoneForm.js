import { TextField } from '@cmsgov/design-system-core';
import PropTypes from 'prop-types';
import React, { Fragment, useCallback } from 'react';
import { connect } from 'react-redux';

import {
  setMilestoneEndDate,
  setMilestoneName
} from '../../../actions/editActivity';
import DateField from '../../../components/DateField';

const MilestoneForm = ({
  activityIndex,
  index,
  item: { endDate, milestone },
  setEndDate,
  setName
}) => {
  const changeDate = useCallback(
    (_, dateStr) => setEndDate(activityIndex, index, dateStr),
    []
  );
  const changeName = useCallback(
    ({ target: { value } }) => setName(activityIndex, index, value),
    []
  );

  return (
    <Fragment>
      <h6 className="ds-h4">Milestone {index + 1}:</h6>
      <TextField
        autoFocus="true"
        label="Name"
        name="name"
        value={milestone}
        onChange={changeName}
      />
      <DateField
        label="Planned end date"
        hint=""
        value={endDate}
        onChange={changeDate}
      />
    </Fragment>
  );
};

MilestoneForm.propTypes = {
  activityIndex: PropTypes.number.isRequired,
  index: PropTypes.number.isRequired,
  item: PropTypes.shape({
    endDate: PropTypes.string.isRequired,
    milestone: PropTypes.string.isRequired
  }).isRequired,
  setEndDate: PropTypes.func.isRequired,
  setName: PropTypes.func.isRequired
};

const mapDispatchToProps = {
  setEndDate: setMilestoneEndDate,
  setName: setMilestoneName
};

export default connect(
  null,
  mapDispatchToProps
)(MilestoneForm);

export { MilestoneForm as plain, mapDispatchToProps };
