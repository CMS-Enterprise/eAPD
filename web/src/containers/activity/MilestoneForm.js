import { Button, TextField } from '@cmsgov/design-system-core';
import PropTypes from 'prop-types';
import React, { useCallback } from 'react';

import DateField from '../../components/DateField';

const MilestoneForm = ({
  idx,
  endDate,
  onChangeDate,
  onChangeName,
  name,
  collapse
}) => {
  const changeDate = useCallback(
    (_, dateStr) => onChangeDate(idx, dateStr),
    []
  );
  const changeName = useCallback(
    ({ target: { value } }) => onChangeName(idx, value),
    []
  );

  return (
    <div className="ds-u-border-bottom--2 ds-u-padding-y--2">
      <h6 className="ds-h4">Milestone {idx + 1}:</h6>
      <TextField label="Name" name="name" value={name} onChange={changeName} />
      <DateField
        label="Planned end date"
        hint=""
        value={endDate}
        onChange={changeDate}
      />
      <Button
        variation="primary"
        className="ds-u-margin-top--2"
        onClick={collapse}
      >
        Done
      </Button>
    </div>
  );
};

MilestoneForm.propTypes = {
  idx: PropTypes.number.isRequired,
  endDate: PropTypes.string.isRequired,
  onChangeDate: PropTypes.func.isRequired,
  onChangeName: PropTypes.func.isRequired,
  name: PropTypes.string.isRequired,
  collapse: PropTypes.func.isRequired
};

export default MilestoneForm;
