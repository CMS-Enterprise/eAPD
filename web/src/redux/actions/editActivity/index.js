export * from './activityList';
export * from './conditionsForEnhancedFunding';
export * from './alternativesAndRisks';
export * from './contractorResources';
export * from './costAllocate';
export * from './nonPersonnelCosts';
export * from './outcomes';
export * from './overview';
export * from './scheduleAndMilestones';
export * from './standardsAndConditions';
export * from './statePersonnel';
