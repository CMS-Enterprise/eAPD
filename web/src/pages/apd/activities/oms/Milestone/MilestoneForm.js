import { TextField } from '@cmsgov/design-system';
import PropTypes from 'prop-types';
import React, { useEffect, useReducer, forwardRef } from 'react';
import { connect } from 'react-redux';

import { useForm, Controller } from 'react-hook-form';
import { joiResolver } from '@hookform/resolvers/joi';

import DateField from '../../../../../components/DateField';

import milestonesSchema from '../../../../../static/schemas/milestones';

import { saveMilestone as actualSaveMilestone } from '../../../../../redux/actions/editActivity';

const MilestoneForm = forwardRef(
  ({ activityIndex, index, item, saveMilestone, setFormValid }, ref) => {
    MilestoneForm.displayName = 'MilestoneForm';

    const {
      control,
      formState: { errors, isValid },
      getFieldState,
      getValues,
      setValue
    } = useForm({
      defaultValues: {
        ...item
      },
      mode: 'onBlur',
      reValidateMode: 'onBlur',
      resolver: joiResolver(milestonesSchema)
    });

    const changeDate = (_, dateStr) =>
      setValue(`endDate`, dateStr);

    const changeName = ({ target: { value } }) =>
      setValue(`milestone`, value);

    useEffect(() => {
      console.log("something changed")
      setFormValid(isValid);
    }, [isValid, errors]); // eslint-disable-line react-hooks/exhaustive-deps

    const onSubmit = e => {
      e.preventDefault();
      saveMilestone(activityIndex, index, getValues());
    };

    return (
      <form index={index} onSubmit={onSubmit}>
        <h6 className="ds-h4">Milestone {index + 1}:</h6>
        <Controller
          control={control}
          name="milestone"
          render={({ field: { value, ...props } }) => (
            <TextField
              {...props}
              data-cy={`milestone-${index}`}
              label="Name"
              name="milestone"
              value={value}
              className="remove-clearfix textfield__container"
              onChange={changeName}
              errorMessage={errors?.milestone?.message}
              errorPlacement="bottom"
            />
          )} 
        />
        <Controller
          name="endDate"
          control={control}
          render={({
            field: { onChange, onBlur, ...props },
            formState: { isTouched }
          }) => (
            <DateField
              {...props}
              isTouched={isTouched}
              label="Target completion date"
              onChange={(e, dateStr) => onChange(dateStr)}
              onComponentBlur={() => {
                onBlur();
                if (getFieldState('end').isTouched) {
                  trigger('end');
                }
              }}
              errorMessage={errors?.endDate?.message}
              errorPlacement="bottom"
            />
          )}
        />
        <input
          className="ds-u-visibility--hidden"
          type="submit"
          ref={ref}
          hidden
        />
      </form>
    );
  }
);

MilestoneForm.propTypes = {
  activityIndex: PropTypes.number.isRequired,
  index: PropTypes.number.isRequired,
  item: PropTypes.shape({
    endDate: PropTypes.string.isRequired,
    milestone: PropTypes.string.isRequired
  }).isRequired,
  saveMilestone: PropTypes.func.isRequired,
  setFormValid: PropTypes.func.isRequired
};

const mapDispatchToProps = {
  saveMilestone: actualSaveMilestone
};

export default connect(null, mapDispatchToProps, null, { forwardRef: true })(
  MilestoneForm
);

export { MilestoneForm as plain, mapDispatchToProps };