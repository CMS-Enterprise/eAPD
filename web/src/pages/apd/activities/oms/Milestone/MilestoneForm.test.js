import { mount } from 'enzyme';
import React from 'react';

import { plain as MilestoneForm, mapDispatchToProps } from './MilestoneForm';

import { saveMilestone as actualSaveMilestone } from '../../../../../redux/actions/editActivity';

describe('the MilestoneForm component', () => {
  const saveMilestone = jest.fn();

  const component = mount(
    <MilestoneForm
      activityIndex={225}
      index={3252}
      item={{
        // Operation Torch, the Allied invasion of North Africa, is launched
        // to relieve pressure on Egypt and provide an invasion route into
        // southern Europe,
        endDate: '1942-8-16',
        milestone: 'Milestone name'
      }}
      saveMilestone={saveMilestone}
    />
  );

  beforeEach(() => {
    saveMilestone.mockClear();
  });

  test('renders correctly', () => {
    expect(component).toMatchSnapshot();
  });

  describe('events', () => {
    test('handles saving the milestone', () => {
      component.find('form').simulate('submit');
      expect(saveMilestone).toHaveBeenCalled();
    });
  });

  it('maps dispatch actions to props', () => {
    expect(mapDispatchToProps).toEqual({
      saveMilestone: actualSaveMilestone
    });
  });
});
