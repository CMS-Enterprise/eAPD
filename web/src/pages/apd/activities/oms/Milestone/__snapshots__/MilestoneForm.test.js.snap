// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the MilestoneForm component renders correctly 1`] = `
<MilestoneForm
  activityIndex={225}
  index={3252}
  item={
    Object {
      "endDate": "1942-8-16",
      "milestone": "Milestone name",
    }
  }
  saveMilestone={[MockFunction]}
>
  <form
    index={3252}
    onSubmit={[Function]}
  >
    <h6
      className="ds-h4"
    >
      Milestone 
      3253
      :
    </h6>
    <TextField
      className="remove-clearfix textfield__container"
      data-cy="milestone-3252"
      label="Name"
      name="name"
      onChange={[Function]}
      type="text"
      value="Milestone name"
    >
      <FormControl
        className="ds-u-clearfix remove-clearfix textfield__container"
        component="div"
        label="Name"
        labelComponent="label"
        render={[Function]}
      >
        <div
          className="ds-u-clearfix remove-clearfix textfield__container"
        >
          <FormLabel
            component="label"
            errorId="field_1-error"
            fieldId="field_1"
            id="field_1-label"
          >
            <label
              className="ds-c-label"
              htmlFor="field_1"
              id="field_1-label"
            >
              <span
                className=""
              >
                Name
              </span>
            </label>
          </FormLabel>
          <TextInput
            data-cy="milestone-3252"
            errorId="field_1-error"
            errorPlacement="top"
            id="field_1"
            name="name"
            onChange={[Function]}
            setRef={[Function]}
            type="text"
            value="Milestone name"
          >
            <input
              aria-invalid={false}
              className="ds-c-field"
              data-cy="milestone-3252"
              id="field_1"
              name="name"
              onChange={[Function]}
              type="text"
              value="Milestone name"
            />
          </TextInput>
        </div>
      </FormControl>
    </TextField>
    <DateField
      errorMessage=""
      hint=""
      label="Target completion date"
      onChange={[Function]}
      onComponentBlur={[Function]}
      value="1942-8-16"
    >
      <DateField
        dateFormatter={[Function]}
        dayInvalid={false}
        dayLabel="Day"
        dayName="day"
        dayValue={16}
        errorMessage=""
        errorPlacement="bottom"
        hint="For example: 4 / 28 / 1986"
        label="Date"
        monthInvalid={false}
        monthLabel="Month"
        monthName="month"
        monthValue={8}
        onBlur={[Function]}
        onChange={[Function]}
        onComponentBlur={[Function]}
        yearInvalid={false}
        yearLabel="Year"
        yearName="year"
        yearValue={1942}
      >
        <FormControl
          component="fieldset"
          errorMessage=""
          errorPlacement="bottom"
          hint="For example: 4 / 28 / 1986"
          label="Date"
          labelComponent="legend"
          render={[Function]}
        >
          <fieldset
            className="ds-c-fieldset"
          >
            <FormLabel
              component="legend"
              errorId="field_2-error"
              errorMessage=""
              hint="For example: 4 / 28 / 1986"
              id="field_2-label"
            >
              <legend
                className="ds-c-label"
                id="field_2-label"
              >
                <span
                  className=""
                >
                  Date
                </span>
                <span
                  className="ds-c-field__hint"
                >
                  For example: 4 / 28 / 1986
                </span>
              </legend>
            </FormLabel>
            <DateInput
              dateFormatter={[Function]}
              dayInvalid={false}
              dayLabel="Day"
              dayName="day"
              dayValue={16}
              labelId="field_2-label"
              monthInvalid={false}
              monthLabel="Month"
              monthName="month"
              monthValue={8}
              onBlur={[Function]}
              onChange={[Function]}
              onComponentBlur={[Function]}
              yearInvalid={false}
              yearLabel="Year"
              yearName="year"
              yearValue={1942}
            >
              <div
                className="ds-c-datefield__container ds-l-form-row"
              >
                <TextField
                  aria-describedby="field_2-label"
                  aria-invalid={false}
                  className="ds-l-col--auto"
                  fieldClassName="ds-c-field--month"
                  inputRef={[Function]}
                  label="Month"
                  labelClassName="ds-c-datefield__label"
                  name="month"
                  numeric={true}
                  onBlur={[Function]}
                  onChange={[Function]}
                  type="text"
                  value={8}
                >
                  <FormControl
                    className="ds-u-clearfix ds-l-col--auto"
                    component="div"
                    inputRef={[Function]}
                    label="Month"
                    labelClassName="ds-c-datefield__label"
                    labelComponent="label"
                    render={[Function]}
                  >
                    <div
                      className="ds-u-clearfix ds-l-col--auto"
                    >
                      <FormLabel
                        className="ds-c-datefield__label"
                        component="label"
                        errorId="field_3-error"
                        fieldId="field_3"
                        id="field_3-label"
                      >
                        <label
                          className="ds-c-label ds-c-datefield__label"
                          htmlFor="field_3"
                          id="field_3-label"
                        >
                          <span
                            className=""
                          >
                            Month
                          </span>
                        </label>
                      </FormLabel>
                      <TextInput
                        aria-describedby="field_2-label"
                        aria-invalid={false}
                        errorId="field_3-error"
                        errorPlacement="top"
                        fieldClassName="ds-c-field--month"
                        id="field_3"
                        name="month"
                        numeric={true}
                        onBlur={[Function]}
                        onChange={[Function]}
                        setRef={[Function]}
                        type="text"
                        value={8}
                      >
                        <input
                          aria-describedby="field_2-label"
                          aria-invalid={false}
                          className="ds-c-field ds-c-field--month"
                          id="field_3"
                          inputMode="numeric"
                          name="month"
                          onBlur={[Function]}
                          onChange={[Function]}
                          pattern="[0-9]*"
                          type="text"
                          value={8}
                        />
                      </TextInput>
                    </div>
                  </FormControl>
                </TextField>
                <span
                  className="ds-c-datefield__separator"
                >
                  /
                </span>
                <TextField
                  aria-describedby="field_2-label"
                  aria-invalid={false}
                  className="ds-l-col--auto"
                  fieldClassName="ds-c-field--day"
                  inputRef={[Function]}
                  label="Day"
                  labelClassName="ds-c-datefield__label"
                  name="day"
                  numeric={true}
                  onBlur={[Function]}
                  onChange={[Function]}
                  type="text"
                  value={16}
                >
                  <FormControl
                    className="ds-u-clearfix ds-l-col--auto"
                    component="div"
                    inputRef={[Function]}
                    label="Day"
                    labelClassName="ds-c-datefield__label"
                    labelComponent="label"
                    render={[Function]}
                  >
                    <div
                      className="ds-u-clearfix ds-l-col--auto"
                    >
                      <FormLabel
                        className="ds-c-datefield__label"
                        component="label"
                        errorId="field_4-error"
                        fieldId="field_4"
                        id="field_4-label"
                      >
                        <label
                          className="ds-c-label ds-c-datefield__label"
                          htmlFor="field_4"
                          id="field_4-label"
                        >
                          <span
                            className=""
                          >
                            Day
                          </span>
                        </label>
                      </FormLabel>
                      <TextInput
                        aria-describedby="field_2-label"
                        aria-invalid={false}
                        errorId="field_4-error"
                        errorPlacement="top"
                        fieldClassName="ds-c-field--day"
                        id="field_4"
                        name="day"
                        numeric={true}
                        onBlur={[Function]}
                        onChange={[Function]}
                        setRef={[Function]}
                        type="text"
                        value={16}
                      >
                        <input
                          aria-describedby="field_2-label"
                          aria-invalid={false}
                          className="ds-c-field ds-c-field--day"
                          id="field_4"
                          inputMode="numeric"
                          name="day"
                          onBlur={[Function]}
                          onChange={[Function]}
                          pattern="[0-9]*"
                          type="text"
                          value={16}
                        />
                      </TextInput>
                    </div>
                  </FormControl>
                </TextField>
                <span
                  className="ds-c-datefield__separator"
                >
                  /
                </span>
                <TextField
                  aria-describedby="field_2-label"
                  aria-invalid={false}
                  className="ds-l-col--auto"
                  fieldClassName="ds-c-field--year"
                  inputRef={[Function]}
                  label="Year"
                  labelClassName="ds-c-datefield__label"
                  name="year"
                  numeric={true}
                  onBlur={[Function]}
                  onChange={[Function]}
                  type="text"
                  value={1942}
                >
                  <FormControl
                    className="ds-u-clearfix ds-l-col--auto"
                    component="div"
                    inputRef={[Function]}
                    label="Year"
                    labelClassName="ds-c-datefield__label"
                    labelComponent="label"
                    render={[Function]}
                  >
                    <div
                      className="ds-u-clearfix ds-l-col--auto"
                    >
                      <FormLabel
                        className="ds-c-datefield__label"
                        component="label"
                        errorId="field_5-error"
                        fieldId="field_5"
                        id="field_5-label"
                      >
                        <label
                          className="ds-c-label ds-c-datefield__label"
                          htmlFor="field_5"
                          id="field_5-label"
                        >
                          <span
                            className=""
                          >
                            Year
                          </span>
                        </label>
                      </FormLabel>
                      <TextInput
                        aria-describedby="field_2-label"
                        aria-invalid={false}
                        errorId="field_5-error"
                        errorPlacement="top"
                        fieldClassName="ds-c-field--year"
                        id="field_5"
                        name="year"
                        numeric={true}
                        onBlur={[Function]}
                        onChange={[Function]}
                        setRef={[Function]}
                        type="text"
                        value={1942}
                      >
                        <input
                          aria-describedby="field_2-label"
                          aria-invalid={false}
                          className="ds-c-field ds-c-field--year"
                          id="field_5"
                          inputMode="numeric"
                          name="year"
                          onBlur={[Function]}
                          onChange={[Function]}
                          pattern="[0-9]*"
                          type="text"
                          value={1942}
                        />
                      </TextInput>
                    </div>
                  </FormControl>
                </TextField>
              </div>
            </DateInput>
          </fieldset>
        </FormControl>
      </DateField>
    </DateField>
    <input
      className="ds-u-visibility--hidden"
      hidden={true}
      type="submit"
    />
  </form>
</MilestoneForm>
`;
